experimental/devices/unicornhd/unicornhd_test.go

```diff
@@ -0,0 +1,247 @@
+package unicornhd
+
+import (
+	"bytes"
+	"errors"
+	"image"
+	"image/color"
+	"testing"
+
+	"periph.io/x/periph/conn/physic"
+	"periph.io/x/periph/conn/spi"
+	"periph.io/x/periph/conn/spi/spitest"
+)
+
+func TestNewFailsWhenConnectionToSpiFails(t *testing.T) {
+	if dev, err := NewUnicornhd(&spiFail{}); dev != nil || err == nil {
+		t.Fatal()
+	}
+}
+
+func TestNewReturnsDriverWithGoodSpi(t *testing.T) {
+	if dev, err := NewUnicornhd(spitest.NewRecordRaw(nil)); dev == nil || err != nil {
+		t.Fatal()
+	}
+}
+
+func TestStringIsDriverNameWidthHeight(t *testing.T) {
+	expectedString := "UnicornHD{16, 16}"
+	dev, _ := NewUnicornhd(spitest.NewRecordRaw(nil))
+	devString := dev.String()
```

Comment:
You can inline this as:

    if devString := dev.String(); devString != expectedString {